### PR TITLE
feature: prevent component update if image reference unchanged

### PR DESCRIPTION
**Pull request by yerzham**
## Feature or Problem
Skip component update if the currently running component for the component id has the same image reference.

If not skipped, `component_scaled` event is emitted asserting that the component with the new image reference is started, then another `component_scaled` event that asserts a component with the same id and image reference is scaled to 0. Thus, it is difficult to tell if it scaled to 0 or was actually updated with an image of the same reference.

## Related Issues
#1972 

## Release Information
next wasmCloud v1 minor or patch release

## Consumer Impact
- Updating a host's running component with the same image reference in `wash-cli` results in an error.
- If wasmCloud host receives a request to update a component and the component image reference did not change, the component update process is skipped
- Changed wash output messages for component update command

### Unit Test(s)
No changes to unit testing

### Acceptance or Integration
- Enabled `integration_update_actor_serial` test for `wash-cli`
- Modified `integration_update_actor_serial` to check an update with the same image reference

### Manual Verification
Start wasmCloud and:
1. `wash start component wasmcloud.azurecr.io/http-hello-world:0.1.0 test`
2. `wash update component test wasmcloud.azurecr.io/http-hello-world:0.1.0`
3. Observe wasmCloud host logs and see `INFO wasmcloud_host::wasmbus: component already updated component_id=testnew_component_ref=wasmcloud.azurecr.io/http-hello-world:0.1.0`

### CODE REVIEW

**Comment by yerzham** [comment]
I can also add checks in `wash-cli` for this, and maybe add some tests for this situation

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
Great contribution to `wash`! I just had a request to move some of the logic from the background task to the `handle_update_actor` function so we can proactively return the error to the user.

File: crates/host/src/wasmbus/mod.rs [hunk 1]
```diff
@@ -1864,6 +1864,11 @@ impl Host {
             let actor = actors.get(component_id).context("component not found")?;
             let annotations = annotations.unwrap_or_default().into_iter().collect();
 
+            if actor.image_reference == new_component_ref {
+                info!(%component_id, %new_component_ref, "component already updated");
+                return Ok(());
+            }
```
[the remark this answers is not in the document]
Comment (yerzham) [reply crates/host/src/wasmbus/mod.rs:1903]: I have kept it because it still makes sense for the task. Let me know if you have other opinions.

**Comment by yerzham** [comment]
I can rewrite the command output messages to follow a certain pattern. 

Regarding command outputs, if the host responds with success set to true even if the component is already updated, should the wash command result return an error or success from its "validation" before the update? Based on that, I can change the update command test to expect an error or success for the added repeated update.

**Review by brooksmtownsend — APPROVED** [review]
Very nice 👍🏻

**Comment by brooksmtownsend** [comment]
> I can rewrite the command output messages to follow a certain pattern.
> 
> Regarding command outputs, if the host responds with success set to true even if the component is already updated, should the wash command result return an error or success from its "validation" before the update? Based on that, I can change the update command test to expect an error or success for the added repeated update.

Good question. I think in this case it's good for `wash` to be able to look at the message, since that's always provided as extra information about the request. I think that an "already updated" result is a success, and makes that control interface command a little more idempotent which is really really nice from the client side.

I think this logic all looks good, are there more changes you'd like to make here before i merge @yerzham ?

**Comment by yerzham** [comment]
Let's double-check again because the response does not appear to answer my question directly (@brooksmtownsend),

Let's say the component is already updated. Currently, it errors to the shell [here](https://github.com/wasmCloud/wasmCloud/pull/1978/files#diff-949a749e93b49be53932b7dab32fe34494b411e56567ebb7504ebd587ea1854cR83-R88) after "validation". But if that validation did not exist or worked incorrectly, it would succeed to the shell around [here](https://github.com/wasmCloud/wasmCloud/pull/1978/files#diff-949a749e93b49be53932b7dab32fe34494b411e56567ebb7504ebd587ea1854cR96-R104) because host sends a successful "already updated" response.

If the component is already updated, the test I added expects an error to the shell by wash. If the validation did not exist and the host's successful "already updated" is received and sent to shell output, the test will fail.

In short, the command line user will receive an error in the shell. I want to make it clear in case somebody writes a shell script and expects the command to start failing on repeated updates. Changing this behavior would be a breaking change.

If you agree that all this makes sense, I do not have anything to add and this can be merged 👍

**Comment by brooksmtownsend** [comment]
Ah I see what you mean @yerzham , then yes I think your update makes sense to ensure that an already-updated component returns a successful message on the CLI side. Once that's in we can merge 👍🏻

**Review by brooksmtownsend — APPROVED** [review]
